core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -69,21 +70,49 @@ class SnappySessionState(snappySession: SnappySession)
 
   private[sql] var disableStoreOptimizations : Boolean = false
 
+  lazy val analyzerPrepare: Analyzer = new Analyzer(catalog, conf) {
```

Comment:
Can you please put a comment on top that what all rules you are skipping and why. Like Promotion rules etc.